### DISCUSSION TITLE
Grouping analytics for dynamic routes

### DISCUSSION THREAD

**Discussion by taylorlapeyre** [plausible/analytics#855] [Anything Else]
Hey there! We'd like to group together page views for "any user profile page" on our site. For instance,

```
Top Visits
/                                        138
/profile/:username                        40
```

I'd love to be able to do this for a bunch of pages on our site with dynamic routes! What do you think?

**Update:** You can group pages [using the "filter" button on top of your dashboard](https://plausible.io/docs/filters-segments#using-the-add-filter-menu-for-extra-functionality).

You can also group pages and have these groups displayed permanently in your dashboard in the Goals section using [pageview goals](https://plausible.io/docs/pageview-goals#pageview-goals-support-wildcards).

We allow you to specify a custom location for your URLs and permanently group pages in your stats dashboard that way. See https://plausible.io/docs/custom-locations

**Comment by Vigasaurus** [comment]
And you can! https://plausible.io/docs/pageview-goals#pageview-goals-support-wildcards

**Reply by taylorlapeyre** [reply]
Amazing! thank you

**Reply by taylorlapeyre** [reply]
I still think a nice thing to add would be to have this support outside of the "Goals" system — maybe a separate section in the site settings where you can list all of your dynamic routes so that they are listed differently on "Top Pages" and "Entry pages" etc

**Reply by ukutaht** [reply]
Yes I think that would be useful as well

**Comment by thefedoration** [comment]
+1 to this, ability to auto-detect or configure patterns of URLs would make it a lot easier to understand user activity.
If that's not easily done, then the ability to report on page title (in addition to url) in the `Top Pages` section would achieve a similar outcome.

I understand this is possible in the Goals, but I'm referring to `Top/Entry/Exit` pages specifically.

**Reply by metmarkosaric** [reply]
Thanks @thefedoration!

You can group pages [using the "filter" button on top of your dashboard](https://plausible.io/docs/filters-segments#using-the-add-filter-menu-for-extra-functionality).

You can also group pages and have these groups displayed permanently in your dashboard in the Goals section using [pageview goals](https://plausible.io/docs/pageview-goals#pageview-goals-support-wildcards).

We allow you to specify a custom location for your URLs and permanently group pages in your stats dashboard that way. See https://plausible.io/docs/custom-locations

And if you do this in the goals section and filter by it, then

**Reply by tavoyne** [reply]
> You can group pages [using the "filter" button on top of your dashboard](https://plausible.io/docs/filters-segments#using-the-add-filter-menu-for-extra-functionality).

I wouldn't say that this lets you group pages, but rather allows you to view the traffic for some pages matching a wildcard in isolation.

The OP wants to be able to see something like this in the Top Pages section:
```
/                                        138
/profile/:username                        40
```

As of today, this is impossible, right?

**Reply by metmarkosaric** [reply]
@tavoyne you can group pages any way that you wish using the "filter" button but you cannot change the page paths permanently in the Top Pages reports using that method. filter button is more meant for stats analysis and the Top Pages report records page paths as they are on your site. if you want to change how page paths of your site are recorded in the Top Pages report, you could use this instead https://plausible.io/docs/custom-locations

**Reply by tavoyne** [reply]
@metmarkosaric I believe what the OP means, as well as @thefedoration, is that it would be great to have the ability to customize what gets displayed in the Top Pages section without necessarily making permanent changes to the page paths. Instead, this customization could be achieved by setting some rules. For example, one possible rule could be to take all page paths that match the `/profile/*` wildcard, group them, and display them under a single `/profile/view` label, **alongside other pages**.

**Reply by metmarkosaric** [reply]
@taylorlapeyre that's not possible unfortunately without making changes to the page paths. you can use the "filter" button or the "pageview goals" functionality to do this but that won't be a permanent change to the Top Pages report. with the "pageview goals" functionality, you'll get this grouping permanently in your dashboard but in the Goals section
